src/System.Linq/tests/ToHashSetTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class ToHashSetTests
+    {
+        private class CustomComparer<T> : IEqualityComparer<T>
+        {
+            public bool Equals(T x, T y) { return EqualityComparer<T>.Default.Equals(x, y); }
+            public int GetHashCode(T obj) { return EqualityComparer<T>.Default.GetHashCode(obj); }
+            public override bool Equals(object obj)
+            {
+                return obj is CustomComparer<T>;
+            }
+            public override int GetHashCode()
+            {
+                return -465576218; // Equals all instances of same class, so randomly-produced constant.
+            }
+        }
+        [Fact]
+        public void NoExplicitComparer()
+        {
+            var hs = Enumerable.Range(0, 50).ToHashSet();
+            Assert.IsType<HashSet<int>>(hs);
+            Assert.Equal(50, hs.Count);
+            Assert.Equal(EqualityComparer<int>.Default, hs.Comparer);
+        }
+        [Fact]
+        public void ExplicitComparer()
+        {
+            var hs = Enumerable.Range(0, 50).ToHashSet(new CustomComparer<int>());
+            Assert.IsType<HashSet<int>>(hs);
+            Assert.Equal(50, hs.Count);
+            Assert.Equal(new CustomComparer<int>(), hs.Comparer);
+        }
+        [Fact]
+        public void TolerateNullElements()
+        {
+            // Unlike the keys of a dictionary, HashSet tolerates null items.
+            new string[] { "abc", null, "def" }.ToHashSet();
```

Comment:
Yes, but not available when I go to build. The change I've made since is nicer still anyway, as it checks on the state of the produced set, along with verifying that it isn't an always-true state.